src/Microsoft.AspNetCore.Mvc.TagHelpers/FormTagHelper.cs

```diff
@@ -201,6 +211,13 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                         routeValues,
                         method: null,
                         htmlAttributes: null);
+
+                    //Append the fragment to action
+                    if (Fragment != null)
+                    {
+                        tagBuilder.Attributes["action"] += string.Concat("#" + Fragment);
```

Comment:
Oh gosh just use `+`. It's super hard to read string interpolation when there's almost nothing being interpolated.